src/utils/schema/widgetPropertyUpdaters.ts

```diff
@@ -0,0 +1,136 @@
+ * @param value The value to set to this key
+ * @param verna The verna context, used to update the rendered schema
+ * @param widgetPath The decomposed id for the corresponding widget
+ */
+function updateProperty(
+  propertyKey: keyof VernaJSONSchemaType,
+  value: unknown,
+  verna: VernaContextProps,
+  widgetPath: string[],
+) {
+  const newSchema = { ...verna.schema };
+  if (verna.selector) {
+    const widget = newSchema?.properties?.[widgetPath[0]];
+    if (widget) {
+      // @ts-ignore
+      widget[propertyKey] = value as number;
+    }
+  } else {
+    const section = newSchema.properties?.[widgetPath[0]];
+    const widget = section?.properties?.[widgetPath[1]];
+    if (widget) {
+      // @ts-ignore
+      widget[propertyKey] = value as number;
+    }
+  }
+  verna.setSchema(newSchema);
+}
+
+/**
+ * Add or remove a widget from required fields inside its section
+ *
+ * @param value The value to add or removed from required property
+ * @param verna The verna context, used to update the rendered schema
+ * @param widgetPath The decomposed id for the corresponding widget
+ */
+function updateRequired(
+  value: Properties['required'],
+  verna: VernaContextProps,
+  widgetPath: string[],
+) {
+  const newSchema = { ...verna.schema };
+  if (verna.selector) {
+    if (value) {
+      newSchema.required = [...(newSchema.required || []), widgetPath[0]];
+    } else {
+      newSchema.required = newSchema.required?.filter((e: string) => e !== widgetPath[0]);
+    }
+  } else {
+    const section = newSchema.properties?.[widgetPath[0]];
+    const widgetName = widgetPath[1];
+    if (section) {
+      if (value) {
+        if (!section.required?.includes(widgetName)) {
+          section.required = [...(section.required || []), widgetName];
+        }
+      } else {
+        section.required = section.required?.filter((e: string) => e !== widgetName);
+      }
+    }
+  }
+  verna.setSchema(newSchema);
+}
+
+/**
+ * Add or remove items from a widget.
+ * The items array is the list of props displayed in the widget
+ * (e.g: a Select field)
+ *
+ * @param value The value to add or removed from required property
+ * @param verna The verna context, used to update the rendered schema
+ * @param widgetPath The decomposed id for the corresponding widget
+ */
+function updateItems(value: Properties['items'], verna: VernaContextProps, widgetPath: string[]) {
+  const newSchema = verna.schema;
+  if (verna.selector) {
+    const widget = newSchema.properties?.[widgetPath[0]];
+    if (widget && widget.items) {
+      (widget.items as VernaJSONSchemaType).enum = value as string[];
+    }
+  } else {
+    const section = newSchema.properties?.[widgetPath[0]];
+    if (section) {
+      const widget = section.properties?.[widgetPath[1]];
+      if (widget && widget.items) {
+        (widget.items as VernaJSONSchemaType).enum = value as string[];
+      }
+    }
+  }
+  verna.setSchema(newSchema);
```

Comment:
What do you think to write things like this to don't repeat yourself. It could be applied to function above.

```suggestion
  const newSchema = verna.schema;
  let widget;

  if (verna.selector) {
    widget = newSchema.properties?.[widgetPath[0]];
  } else {
    const section = newSchema.properties?.[widgetPath[0]];
    widget = section?.properties?.[widgetPath[1]];
  }
  
    if (widget && widget.items) {
      (widget.items as VernaJSONSchemaType).enum = value as string[];
      verna.setSchema(newSchema);
    }
```